File: ZAD_1/app/src/main/java/com/example/zad_1/MainActivity.java
package com.example.zad_1;

import androidx.appcompat.app.AppCompatActivity;
import androidx.navigation.NavDirections;
import androidx.navigation.Navigation;

import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import java.text.NumberFormat;

public class MainActivity extends AppCompatActivity {
    static int kcal = -1;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

    }


    public void actionMainToFragmentBmi(View view){
        NavDirections action = FragmentMainDirections.actionMainToFragmentBmi();
        Navigation.findNavController(view).navigate(action);
    }
    public void actionMainToFragmentKcal(View view){
        NavDirections action = FragmentMainDirections.actionMainToFragmentKcal();
        Navigation.findNavController(view).navigate(action);
    }
    public void actionKcalToFragmentWhatEat(View view){
        NavDirections action = FragmentKcalDirections.actionFragmentKcalToFragmentWhatEat();
        Navigation.findNavController(view).navigate(action);
    }
    static void setKcal(int kcalIn){
        kcal = kcalIn;
    }
    static int getKcal(){
        return kcal;
    }

}

File: ZAD_1/app/src/main/java/com/example/zad_1/MainActivity.kt
package com.example.zad_1

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.Navigation

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun actionMainToFragmentBmi(view: View?) {
        val action = FragmentMainDirections.actionMainToFragmentBmi()
        if (view != null) {
            Navigation.findNavController(view).navigate(action)
        }
    }

    fun actionMainToFragmentKcal(view: View?) {
        val action = FragmentMainDirections.actionMainToFragmentKcal()
        if (view != null) {
            Navigation.findNavController(view).navigate(action)
        }
    }

    fun actionKcalToFragmentWhatEat(view: View?) {
        val action = FragmentKcalDirections.actionFragmentKcalToFragmentWhatEat()
        if (view != null) {
            Navigation.findNavController(view).navigate(action)
        }
    }

    fun actionMainToQuiz(view: View?) {
        val action = FragmentMainDirections.actionMainToFragmentQuiz()
        if (view != null) {
            Navigation.findNavController(view).navigate(action)
        }
    }

    fun actionMainToChart(view: View?) {
        val action = FragmentMainDirections.actionMainToFragmentChart()
        if (view != null) {
            Navigation.findNavController(view).navigate(action)
        }
    }

    companion object {
        private var kcal = -1
        @JvmName("setKcal1")
        fun setKcal(kcalIn: Int) {
            kcal = kcalIn
        }

        @JvmName("getKcal1")
        fun getKcal(): Int {
            return kcal
        }
    }
}